libraries/base/Data/Monoid.hs

```diff
@@ -16,8 +16,29 @@
 -- Stability   :  experimental
 -- Portability :  portable
 --
--- A class for monoids (types with an associative binary operation that
--- has an identity) with various general-purpose instances.
+-- A type @a@ is a 'Monoid' if it provides an associative function ('<>')
+-- that lets you combine any two values of type @a@ into one, and a neutral
+-- element (`mempty`) such that 
+--
+-- > a <> mempty == mempty <> a == a
+--
+-- ==== __Examples__
+-- >>> Sum 1 <> Sum 2 <> Sum 3 <> Sum 4 :: Sum Int
```

Comment:
I wonder if we should add a sentence briefly introducing `Sum` so it is clear to a new-comer what this example means. For instance,

```
Take, for instance, the example of the 'Sum' monoid, defined by the numerical 
addition operator and the neutral element `0`:
>>> mempty :: Sum Int
Sum 0
>>> Sum 1 <> Sum 2 <> Sum 3 <> Sum 4 :: Sum Int
Sum {getSum = 10}
```
and so on.

Reply:
done, PTAL